workitem/event/event_repository_blackbox_test.go

```diff
@@ -0,0 +1,315 @@
+	})
+
+	s.T().Run("event label - new label nil", func(t *testing.T) {
+
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1))
+
+		label := []string{"label1"}
+
+		fxt.WorkItems[0].Fields[workitem.SystemLabels] = label
+		wiNew, err := s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
+		require.NoError(t, err)
+		require.Len(t, wiNew.Fields[workitem.SystemLabels].([]interface{}), 1)
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 1)
+		assert.Equal(t, eventList[0].Name, workitem.SystemLabels)
+		assert.Empty(t, eventList[0].Old)
+
+		wiNew.Fields[workitem.SystemLabels] = []string{}
+		wiNew.Version = fxt.WorkItems[0].Version + 1
+		wiNew, err = s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *wiNew, fxt.Identities[0].ID)
+		require.NoError(t, err)
+		require.Len(t, wiNew.Fields[workitem.SystemLabels].([]interface{}), 0)
+		eventList, err = s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 2)
+		assert.Equal(t, eventList[1].Name, workitem.SystemLabels)
+		assert.Empty(t, eventList[1].New)
+	})
+
+	s.T().Run("iteration changed", func(t *testing.T) {
+
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1), tf.Iterations(2))
+		fxt.WorkItems[0].Fields[workitem.SystemIteration] = fxt.Iterations[0].ID.String()
+		wiNew, err := s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
+		require.NoError(t, err)
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 1)
+		assert.Equal(t, eventList[0].Name, workitem.SystemIteration)
+		assert.Empty(t, eventList[0].Old)
+
+		wiNew.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+		wiNew.Version = fxt.WorkItems[0].Version + 1
+		wiNew, err = s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *wiNew, fxt.Identities[0].ID)
+		require.NoError(t, err)
+		eventList, err = s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.Len(t, eventList, 2)
+		assert.Equal(t, eventList[1].Name, workitem.SystemIteration)
+	})
+
+	s.T().Run("multiple events", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1))
+		label := []string{"label1"}
+		fxt.WorkItems[0].Fields[workitem.SystemLabels] = label
+		fxt.WorkItems[0].Fields[workitem.SystemState] = workitem.SystemStateResolved
+		_, err := s.wiRepo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
+		require.NoError(t, err)
+		eventList, err := s.wiEventRepo.List(s.Ctx, fxt.WorkItems[0].ID)
+		require.NoError(t, err)
+		require.NotEmpty(t, eventList)
+		require.Len(t, eventList, 2)
+		c := 0
+		for _, k := range eventList {
+			switch k.Name {
+			case workitem.SystemState:
+				c = c + 1
+			case workitem.SystemLabels:
+				c = c + 1
+			}
+		}
+		assert.Equal(t, 2, c)
```

Comment:
@baijum Usually these tests are not cool because if either the state or the labels have two events and the other is missing, this test would still pass.

Reply:
I agree we can improve the test.